src/app.py

```diff
@@ -4772,6 +4777,35 @@ def access_level_prefix_dir(dir_name):
     return hm_file_helper.ensureTrailingSlashURL(hm_file_helper.ensureBeginningSlashURL(dir_name))
 
 
+"""
+Check if a user has valid access to update a given entity
+
+Parameters
+----------
+entity : str
+    The entity that is attempting to be updated
+user_token : str 
+    The token passed in via the request header that will be used to authenticate
+
+"""
+
+
+def validate_user_update_privilege(entity, user_token):
+    # A user has update privileges if they are a data admin or are in the same group that registered the entity
+    is_admin = auth_helper_instance.has_data_admin_privs(user_token)
+    if isinstance(is_admin, Response):
+        is_admin = False
+
+    user_write_groups: List[dict] = auth_helper_instance.get_user_write_groups(user_token)
+    if isinstance(user_write_groups, Response):
+        abort_forbidden(f"Entity with ID: {entity['sennet_id']} is not editable without presenting a token.")
```

Comment:
By the time the request hits to this point, it has already passed the API Gateway read-token check. I think there's no need to send back this error message asking for a token, it won't happen.